Feature/modifica construtor resource

rtakauti opened the pull request:
## Motivação
Atendendo a dois issues apresentados pelo github de usuários do SDK-PHP foi criado a opção de passar VINDI_API_URI e VINDI_API_KEY como argumentos na primeira instância de uma classe filha de Resource, sendo ignorada na tentativa de reinserção em uma nova instância.

## Solução proposta
Implementação da feature que atendesse a expectativa dos usuários de não só depender de variáveis de ambiente para configurar VINDI_API_URI e VINDI_API_KEY e refatoração do código para que ficasse coeso com a solução.

rtakauti committed:
Adiciona pasta de configuração do PHPStorm no .gitignore

rtakauti committed:
Adiciona  opção de passar VINDI_API_URI e VINDI_API_KEY como argumentos

Opção de passar VINDI_API_URI e VINDI_API_KEY como argumentos nas instancias de classes filhas de Resource

wagner commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 build
 composer.lock
 vendor
+.idea/
```
Esse é um arquivo do seu editor? Se for, é melhor colocar no `git config --global` em vez de adicionar no projeto.

wagner commented on README.md:
```diff
@@ -1,4 +1,4 @@
-![alt text align:center](https://www.vindi.com.br/image/vindi-logo-transparente.png "Vindi")
+![alt text align:center](https://www.vindi.com.br/assets/images/logo-footer.png "Vindi")
```
A imagem não apareceu aqui.

rtakauti replied:
Foi uma escolha de imagem branca que não aparecia no fundo de mesma cor.

wagner commented on README.md:
```diff
@@ -42,6 +42,15 @@ putenv('VINDI_API_KEY=SUA_CHAVE_DA_API');
+    'VINDI_API_URI' => 'https://sandbox-app.vindi.com.br/api/v1/'
+);
+$customerService = new Vindi\Customer($arguments);
+
```
Acredito que a documentação não está clara quanto às formas de autenticação.

Minha sugestão seria criar uma seção "Autenticação" explicando que preferencialmente a variável de ambiente será utilizada, ao menos que a chave seja explicitamente configurada na instância.

wagner commented on src/Vindi.php:
```diff
@@ -63,10 +118,10 @@ public static function getApiKey()
     {
-        if (empty(getenv(static::$apiTimeOutEnvVar))) {
-            return 60;
+        if (!empty(getenv(static::$apiTimeOutEnvVar))) {
```
Esse código sobre timeout tem alguma relação com o PR?

wagner commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
+## 1.1.0 - 04/05/2018
+- Option to input VINDI_API_URI passing as arguments on the instances.
+- Option to input VINDI_API_KEY passing as arguments on the instances.
+
```
Dúvida: Esse changelog mistura português e inglês por algum motivo específico? Ele é exibido em algum outro lugar?

rtakauti replied:
O CHANGELOG.md está em inglês, para seguir com o padrão essa nova versão também deixei inglês

wagner reviewed (changes requested):
Eu não consigo falar muito sobre o código PHP, mas a documentação precisa ficar mais clara sobre essa nova funcionalidade.

rtakauti committed:
Atualiza versão 1.1.0 na classe Vindi

rtakauti committed:
Volta o código original

rtakauti committed:
Retira .idea do gitignore

rtakauti committed:
Adiciona  opção de passar VINDI_API_URI e VINDI_API_KEY como argumentos

Opção de passar VINDI_API_URI e VINDI_API_KEY como argumentos nas instâncias de classes filhas de Resource
Refatora código da classe Vindi para melhor adequar as mudanças realizadas nessa feature

rtakauti committed:
Refatora README para melhor entendimento da nova feature

rtakauti committed:
Refatora CHANGELOG para melhor entendimento

carolina-y commented on README.md:
```diff
@@ -66,6 +67,60 @@ foreach ($customers as $customer) {
+
+// Declara em um array os valores de VINDI_API_KEY e VINDI_API_URI
+$arguments = array(
+    \Vindi\Vindi::VINDI_API_KEY => 'SUA_CHAVE_DA_API',
```
A pessoa não pode chamar via este código abaixo?

```
$arguments = array(
    'VINDI_API_KEY' => 'SUA_CHAVE_DA_API',
    'VINDI_API_URI' => 'https://sandbox-app.vindi.com.br/api/v1/'
);
```

(entendo a completude, mas é mais legível para humanos 'VINDI_API_KEY' e afins)

carolina-y commented on src/Vindi.php:
```diff
@@ -4,27 +4,41 @@
      *
      * @var string
      */
-    public static $apiUriEnvVar = 'VINDI_API_URI';
```
:+1:

lyoncesar replied:
Também curti isso aqui :+1:

rtakauti committed:
Altera idioma de uma frase no CHANGELOG

rtakauti committed:
Refatora para melhor entendimento do README

rtakauti committed:
Refatora ordem de importância de VINDI_API_KEY e VINDI_API_URI

rtakauti committed:
Adiciona exemplo de API URI

rtakauti committed:
Refatora teste da classe Vindi

lyoncesar commented on README.md:
```diff
@@ -66,6 +70,60 @@ foreach ($customers as $customer) {
 
+### Exemplo de Autenticação #2
+
+> Esse método de autenticação utiliza-se de inserção de um *array* como argumento na primeira instância de uma classe *filha* de Resource, sendo ignorada uma nova tentativa de inserir o argumento em uma outra instância. 
```
Será que tem como deixar em negrito essa parte do texto avisa sobre a inserção de um novo argumento em outra instância? Acho que isso pode sinalizar a importância do alerta para o cliente.

rtakauti committed:
Altera ênfase na explicação sobre outra instancia

wagner commented on README.md:
````diff
@@ -66,6 +70,60 @@ foreach ($customers as $customer) {
+
+echo "Novo produto criado com o id  '{$product->id}'.<br />";
+```
+
````
O texto foi copiado e colado só mudando o método de autenticação? Não vai dar o dobro de trabalho pra atualizar depois?

Eu havia imaginado simplesmente criar uma seção "Autenticação" com os dois métodos.

wagner reviewed (approved):
Estou aprovando, mas acredito que a documentação poderia ser melhor, evitando a duplicação no exemplo e também incluindo alguma referência para os outros serviços ao invés de manter uma documentação paralela no Zendesk.

rtakauti committed:
Separa os meios de autenticação do README

rtakauti committed:
Retira menção sobre .env

lyoncesar commented:
Acho que está dentro da documentação atual, podemos levantar a documentação de todas as integrações para validar tudo e depois criarmos uma nova.
